server/middleware/assignParking.js

```diff
@@ -0,0 +1,42 @@
+const { ParkingSpots, Reservations } = require('../db.js');
+
+
+// need to do a join table
+// get all parking spots in ascending order
+  // where reservations at that time/date doesn't exist
+
+const assignParking = async (req, res, next) => {
+  const { data: ParkingData, error: ParkingError } = await ParkingSpots
+    .select(`id`)
+    .eq('garage_id', req.body.garage_id)
+    .order('id', { ascending: true })
+
+  const { data: ReservationData, error: ReservationError } = await Reservations
+    .select(`parking_spot_id`)
+    .match({ date: req.body.date, time: req.body.time, garage_id: req.body.garage_id })
+    .order('parking_spot_id', { ascending: true })
+
+  if (ParkingError || ReservationError) {
+    console.log(ParkingError);
+    return res.sendStatus(500);
+  }
+
+  let parkingSpot;
+
+  while (!parkingSpot && ParkingData.length > 0 && ReservationData.length > 0) {
+    const spot = ParkingData.shift();
+    const reservation = ReservationData.shift();
+
+    if (spot.id !== reservation.parking_spot_id) {
+      parkingSpot = spot.id;
+    }
+  }
+
+  if (!parkingSpot) {
+    res.sendStatus(400);
+  }
+  req.body['parking_spot_id'] = parkingSpot;
+  next();
+}
+
+module.exports = assignParking;
```

Comment:
finds the first available parking spot and passes that spot to create a reservation, if no spots available it returns status 400. only consideration would be to alert the user that the parking garage is full so that they are not just getting a 400 error with no message to tell them to find another garage to book in.